File: bingNews/Bing/Models/ImageObject.cs
using Microsoft.Kiota.Abstractions.Serialization;
using System;
using System.Collections.Generic;
using System.IO;
using System.Linq;
namespace Bing.Models {
    /// <summary>Defines an image</summary>
    public class ImageObject : MediaObject, IParsable {
        /// <summary>Defines an image</summary>
        public ImageObject Thumbnail { get; set; }
        /// <summary>
        /// Creates a new instance of the appropriate class based on discriminator value
        /// <param name="parseNode">The parse node to use to read the discriminator value and create the object</param>
        /// </summary>
        public static new ImageObject CreateFromDiscriminatorValue(IParseNode parseNode) {
            _ = parseNode ?? throw new ArgumentNullException(nameof(parseNode));
            return new ImageObject();
        }
        /// <summary>
        /// The deserialization information for the current model
        /// </summary>
        public new IDictionary<string, Action<IParseNode>> GetFieldDeserializers() {
            return new Dictionary<string, Action<IParseNode>>(base.GetFieldDeserializers()) {
                {"thumbnail", n => { Thumbnail = n.GetObjectValue<ImageObject>(ImageObject.CreateFromDiscriminatorValue); } },
            };
        }
        /// <summary>
        /// Serializes information the current object
        /// <param name="writer">Serialization writer to use to serialize this model</param>
        /// </summary>
        public new void Serialize(ISerializationWriter writer) {
            _ = writer ?? throw new ArgumentNullException(nameof(writer));
            base.Serialize(writer);
            writer.WriteObjectValue<ImageObject>("thumbnail", Thumbnail);
        }
    }
}

File: src/js/bingNews/models/imageObject.ts
import {createImageObjectFromDiscriminatorValue} from './createImageObjectFromDiscriminatorValue';
import {MediaObject} from './index';
import {Parsable, ParseNode, SerializationWriter} from '@microsoft/kiota-abstractions';

/** Defines an image */
export class ImageObject extends MediaObject implements Parsable {
    /** Defines an image */
    private _thumbnail?: ImageObject | undefined;
    /**
     * Instantiates a new ImageObject and sets the default values.
     */
    public constructor() {
        super();
    };
    /**
     * The deserialization information for the current model
     * @returns a Record<string, (node: ParseNode) => void>
     */
    public getFieldDeserializers() : Record<string, (node: ParseNode) => void> {
        return {...super.getFieldDeserializers(),
            "thumbnail": n => { this.thumbnail = n.getObjectValue<ImageObject>(createImageObjectFromDiscriminatorValue); },
        };
    };
    /**
     * Serializes information the current object
     * @param writer Serialization writer to use to serialize this model
     */
    public serialize(writer: SerializationWriter) : void {
        if(!writer) throw new Error("writer cannot be undefined");
        super.serialize(writer);
        writer.writeObjectValue<ImageObject>("thumbnail", this.thumbnail);
    };
    /**
     * Gets the thumbnail property value. Defines an image
     * @returns a ImageObject
     */
    public get thumbnail() {
        return this._thumbnail;
    };
    /**
     * Sets the thumbnail property value. Defines an image
     * @param value Value to set for the thumbnail property.
     */
    public set thumbnail(value: ImageObject | undefined) {
        this._thumbnail = value;
    };
}
